### PR TITLE
ratio_of_average_power_to_incident_power_from_reflection_coefficient

### PR DESCRIPTION
**Pull request by Zulusssss**
Add law and tests

### CODE REVIEW

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+# Description
+## The reflection coefficient is equal to the ratio of the reflected wave power to the incident wave power.
+## Knowing the average power delivered to the load and the incident power, it is possible to calculate the absolute value reflection coefficient.
+
+## Law is: Pa / Pi = 1 - G^2, where
+## Pa - the average power delivered to the load,
+## Pi - the incident power,
+## K - the absolute value reflection coefficient.
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:19]: I do not see K in formula above.

Comment (blackyblack) [inline symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:19]: How is it different from normal reflection coefficient?

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+absolute_reflection_coefficient = Symbol("absolute_reflection_coefficient", dimensionless, rel=True)
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:21]: Why `rel=True`?

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+incident_power = Symbol("incident_power", units.power, rel=True)
+average_power = Symbol("average_power", units.power, rel=True)
+
+law = Eq(average_power / incident_power, 1 - absolute_reflection_coefficient**2)
+
+
+@validate_input(incident_power_=incident_power, average_power_=average_power)
+@validate_output(absolute_reflection_coefficient)
+def calculate_absolute_reflection_coefficient(incident_power_: Quantity, average_power_: Quantity) -> float:
+    result_expr = solve(law, absolute_reflection_coefficient, dict=True)[0][absolute_reflection_coefficient]
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:31]: Do we want to check that incident power is not less than average power?

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    result_expr = result_expr.subs({
+        incident_power: incident_power_,
+        average_power: average_power_,
+    })
+    return convert_to_float(result_expr)
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:36]: It might return negative solution.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 1 above)
Comment (Zulusssss) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:19]: > How is it different from normal reflection coefficient?

This is the reflection coefficient module. That's all the difference.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 2 above)
Comment (Zulusssss) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:21]: > Why `rel=True`?

Because the reflection coefficient modulus cannot be imaginary.

Comment (blackyblack) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:20]: It's `real` as far as I remember.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 1 above)
Comment (blackyblack) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py]: But in this formula it does not matter if it is absolute value or not.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py [hunk 5]
```diff
@@ -28,9 +27,11 @@
...
 def calculate_absolute_reflection_coefficient(incident_power_: Quantity, average_power_: Quantity) -> float:
+    if incident_power_.scale_factor < average_power_.scale_factor:
+        raise ValueError("The incident_power must be greater than the average power")
     result_expr = solve(law, absolute_reflection_coefficient, dict=True)[0][absolute_reflection_coefficient]
     result_expr = result_expr.subs({
         incident_power: incident_power_,
         average_power: average_power_,
     })
-    return convert_to_float(result_expr)
+    return result_expr
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:37]: Why this change?

Comment (Zulusssss) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:37]: > Why this change?

Because I didn't understand what was meant by that remark:
https://github.com/blackyblack/symplyphysics/pull/716#discussion_r1580030552

I thought I should delete convert_to_float()

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 2 above)
Comment (Zulusssss) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:20]: > It's `real` as far as I remember.

Is it real that the reflection coefficient module can contain an imaginary part? No, that doesn't happen.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 5 above)
Comment (blackyblack) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:37]: You state that you've got absolute value, but solve() might return negative or positive solution. So I expect you do something with its sign or return just value, not absolute value.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 1 above)
Comment (blackyblack) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py]: Oh, I see. Add some comments to guide future readers on this case.

File: symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py (hunk 5 above)
Comment (blackyblack) [reply symplyphysics/laws/electricity/transmission_lines/ratio_of_average_power_to_incident_power_from_reflection_coefficient.py:37]: As soon as reflection coefficient is complex, its absolute value can be negative. So you shouldn't check the sign. Revert to `convert_to_float`.

**Review by blackyblack — APPROVED** [review]
